demo/src/http_client.rs

```diff
@@ -0,0 +1,13 @@
+use aries_vcx::{agency_client::httpclient::post_message, errors::error::VcxResult, transport::Transport};
+use async_trait::async_trait;
+use url::Url;
+
+pub struct HttpClient;
+
+#[async_trait]
+impl Transport for HttpClient {
+    async fn send_message(&self, msg: Vec<u8>, service_endpoint: Url) -> VcxResult<()> {
+        post_message(msg, service_endpoint).await?;
```

Comment:
Q) should we still be using stuff from the `agency_client` crate? I wonder if the demo would be cleaner if you just redefined `post_message` in this file rather than pulling in agency_client; the `agency_client::post_message` impl is full of stuff that consumers might not care about (e.g. mocking and layers of error handling, etc).

Reply:
There was an initiative to get the message sending part to `shared_vcx` at some point, which would've maybe helped with this. I believe it was some guy Andy who tried working on it and made a PR, but it just dried out I think.

Reply:
While I agree, connection has been freed from message sending IO and demo has been freed of HTTP communications. So resolving this